### ISSUE TITLE
4.0.1 Fresh installation ERROR: I can not find PHP-CLI anywhere!

### ISSUE THREAD

**Issue by hodbakhalaf** [omega8cc/boa#1382]
Hi, 

I am trying to install a fresh installation of BOA 4.0.1 on a Debian 9 installation, AWS EC2 hosting but the installation faild with the error "ERROR: I can not find PHP-CLI anywhere!"

When it failed the first time I edited the installation to add PHP 7.3 but same issue.

Here is the log:
`BOA [09:47:17] ==> Skynet Agent v.BOA-4.0.1 welcomes you aboard!
 
BOA [09:47:20] ==> INFO: NORMAL INIT
BOA [09:47:20] ==> INFO: Public Setup Mode Active
BOA [09:47:23] ==> VIRT: This system is supported: Xen guest fully virtualized (HVM)
 
BOA [09:47:23] ==> Aegir on ip-172-31-43-20 Debian/stretch x86_64 PHP 7.2
 
BOA [09:47:23] ==> INFO: Installing some basic tools now, please wait...
BOA [09:47:36] ==> INFO: Installing sysvinit-core on Debian stretch...

BOA [09:47:46] ==> NOTE: Please reboot and run barracuda upgrade
BOA [09:47:46] ==> NOTE: after boa completes initial installation
BOA [09:47:46] ==> NOTE: in order to remove unused systemd packages!

BOA [09:48:01] ==> INFO: Installing more basic tools now, please wait...
BOA [09:48:30] ==> INFO: DNS test: OK
 
BOA [09:48:30] ==> INSTALL START -> checkpoint: 

    * Your email address is XXXXX
    * Your IP address is XXXXX
    * Your hostname is XXXX
 
BOA [09:48:30] ==> INFO: Installing extra Drush versions...
BOA [09:48:34] ==> INFO: Running system packages update...
BOA [09:49:45] ==> INFO: Installing required libraries and tools, please wait...
BOA [09:50:59] ==> INFO: Installed OpenSSH version 7.4p1, upgrade required
BOA [09:50:59] ==> INFO: Building OpenSSH 8.0p1 from sources, please wait...
BOA [09:53:53] ==> INFO: Installing MariaDB...
BOA [09:54:03] ==> INFO: Retrieving F1656F24C74CD1D8 key..
BOA [09:54:07] ==> INFO: Retrieving 9334A25F8507EFA5 key..
BOA [09:54:09] ==> INFO: Installing MariaDB 10.1.39 in Debian/stretch
BOA [09:55:08] ==> INFO: Running aptitude full-upgrade...
BOA [09:56:58] ==> INFO: Installed Nginx version nginx/1.10.3, upgrade required
BOA [09:56:59] ==> INFO: Installing Nginx...
BOA [09:59:29] ==> INFO: Installing MySecureShell master-20-03-2015...
BOA [10:01:10] ==> INFO: Installing wkhtmltopdf and wkhtmltoimage x86_64...
BOA [10:01:13] ==> INFO: Fix #1 for libs in Debian stretch
BOA [10:01:13] ==> INFO: Fix #2 for libs in Debian stretch
BOA [10:01:13] ==> INFO: Installing VnStat monitor...
BOA [10:01:28] ==> INFO: Installing a few more tools...
BOA [10:01:28] ==> INFO: PHP 7.2.18 will be installed now
BOA [10:01:28] ==> INFO: Building PHP 7.2.18 from sources, please wait...
BOA [10:01:34] ==> INFO: Building PHP 7.2.18 part 1/3
BOA [10:01:38] ==> INFO: Building PHP 7.2.18 part 2/3
BOA [10:02:19] ==> INFO: Building PHP 7.2.18 part 3/3
BOA [10:06:12] ==> WARN: Building PHP 7.2.18 failed!
BOA [10:06:12] ==> INFO: PHP 7.3.5 will be installed now
BOA [10:06:12] ==> INFO: Building PHP 7.3.5 from sources, please wait...
BOA [10:06:19] ==> INFO: Building PHP 7.3.5 part 1/3
BOA [10:06:23] ==> INFO: Building PHP 7.3.5 part 2/3
BOA [10:07:05] ==> INFO: Building PHP 7.3.5 part 3/3
BOA [10:11:15] ==> WARN: Building PHP 7.3.5 failed!
BOA [10:11:15] ==> WAIT: I can not find PHP-CLI anywhere!
BOA [10:14:16] ==> INFO: Installing Limited Shell 0.9.18.9...
BOA [10:14:30] ==> INFO: Installing Redis update for Debian/stretch...
BOA [10:15:27] ==> INFO: Generating random password for MariaDB
BOA [10:15:27] ==> INFO: Updating MariaDB configuration
BOA [10:15:27] ==> INFO: Restarting MariaDB server...
BOA [10:15:39] ==> INFO: Installing Aegir Master Instance, please wait...
BOA [10:17:49] ==> INFO: MariaDB is listening on 3.92.99.11.
BOA [10:17:57] ==> FATAL ERROR: AegirSetupM installer failed
BOA [10:17:57] ==> FATAL ERROR: Aborting Barracuda installer NOW!
BOA [10:17:57] ==> HINT: Please check /var/aegir/install.log
BOA [10:17:57] ==> HINT: for more information on errors occured
 
BOA [10:18:01] ==> Skynet Agent v.BOA-4.0.1 welcomes you aboard!
 
BOA [10:18:04] ==> ERROR: This octopus installer can be used only when the same version
BOA [10:18:04] ==> of boa or barracuda installer was used before. Your system must be
BOA [10:18:04] ==> upgraded with barracuda installer version BOA-4.0.1 first
BOA [10:18:04] ==> Bye`

and here is the /var/aegir/install.log 
`BOA [10:17:55] ==> INFO: Installing drush in /var/aegir
BOA [10:17:57] ==> FATAL ERROR: _PHP_CLI_VERSION must be set to one of supported values
BOA [10:17:57] ==> FATAL ERROR: Aborting AegirSetupM installer NOW!`

I tried to reinstall it again, running barracuda up-stable but I have the same error.

Would you please point what I am missing here.

Thanks

**Comment by hodbakhalaf** [comment]
The server was too small.

**Comment by omega8cc** [comment]
Thanks for the update! It looked quite mysterious indeed.

**Comment by TheRuslan** [comment]
The same here.

```
# boa in-stable public XXX YYY o1

BOA [11:25:34] ==> Skynet Agent v.BOA-4.0.1 welcomes you aboard!

BOA [11:25:37] ==> INFO: NORMAL INIT
BOA [11:25:38] ==> INFO: Public Setup Mode Active
Do you want to proceed with the install? [Y/n] y

BOA [11:25:51] ==> VIRT: This system is supported: Linux KVM guest

BOA [11:25:51] ==> Aegir on XXX Debian/stretch x86_64 PHP 7.2

BOA [11:25:52] ==> INFO: Installing some basic tools now, please wait...
BOA [11:26:14] ==> INFO: Installing sysvinit-core on Debian stretch...

BOA [11:26:24] ==> NOTE: Please reboot and run barracuda upgrade
BOA [11:26:24] ==> NOTE: after boa completes initial installation
BOA [11:26:24] ==> NOTE: in order to remove unused systemd packages!

BOA [11:26:33] ==> INFO: Updating packages sources list...
BOA [11:26:33] ==> INFO: We will use Debian mirror httpredir.debian.org
BOA [11:26:43] ==> INFO: Installing more basic tools now, please wait...
BOA [11:27:20] ==> INFO: Checking BARRACUDA version...
BOA [11:27:20] ==> INFO: BARRACUDA version test: OK
BOA [11:27:23] ==> INFO: DNS test: OK

BOA [11:27:23] ==> INSTALL START -> checkpoint:

    * Your email address is YYY
    * Your IP address is ZZZ
    * Your hostname is XXX

Do you want to proceed with the install? [Y/n] y
BOA [11:27:37] ==> INFO: Cleaning up temp files in /var/opt/
BOA [11:27:37] ==> INFO: Installing extra Drush versions...
BOA [11:27:39] ==> INFO: Drush mini-7-30-06-2015 installation complete
BOA [11:27:40] ==> INFO: Drush 8.2.3.1 installation complete
BOA [11:27:43] ==> INFO: Running system packages update...
BOA [11:29:25] ==> INFO: Installing required libraries and tools, please wait...
BOA [11:30:44] ==> INFO: Installed OpenSSH version 7.4p1, upgrade required
BOA [11:30:44] ==> INFO: Building OpenSSH 8.0p1 from sources, please wait...
BOA [11:33:18] ==> INFO: Installing MariaDB...
BOA [11:33:30] ==> INFO: Retrieving F1656F24C74CD1D8 key..
BOA [11:33:34] ==> INFO: Retrieving 9334A25F8507EFA5 key..
BOA [11:33:36] ==> INFO: Installing MariaDB 10.2.23 in Debian/stretch
BOA [11:36:22] ==> INFO: Running aptitude full-upgrade...
BOA [11:38:33] ==> INFO: Testing Nginx version...
BOA [11:38:33] ==> INFO: Installed Nginx version nginx/1.10.3, upgrade required
BOA [11:38:33] ==> INFO: Installing Nginx...
BOA [11:40:46] ==> INFO: Installing MySecureShell master-20-03-2015...
BOA [11:42:28] ==> INFO: Installing wkhtmltopdf and wkhtmltoimage x86_64...
BOA [11:42:34] ==> INFO: Fix #1 for libs in Debian stretch
BOA [11:42:34] ==> INFO: Fix #2 for libs in Debian stretch
BOA [11:42:34] ==> INFO: Installing VnStat monitor...
BOA [11:42:49] ==> INFO: Installing a few more tools...
BOA [11:42:50] ==> INFO: PHP 7.2.18 will be installed now
BOA [11:42:50] ==> INFO: THIS PHP EXTRA is --with-ldap --with-gmp --with-xpm-dir=/usr
BOA [11:42:50] ==> INFO: Building PHP 7.2.18 from sources, please wait...
BOA [11:42:57] ==> INFO: Building PHP 7.2.18 part 1/3
BOA [11:43:01] ==> INFO: Building PHP 7.2.18 part 2/3
BOA [11:43:43] ==> INFO: Building PHP 7.2.18 part 3/3
BOA [11:49:09] ==> WARN: Building PHP 7.2.18 failed!
BOA [11:49:09] ==> WAIT: I can not find PHP-CLI anywhere!
BOA [11:52:09] ==> INFO: Installing Limited Shell 0.9.18.9...
BOA [11:52:25] ==> INFO: Installing Redis update for Debian/stretch...
BOA [11:53:11] ==> INFO: Generating random password for local Redis server
BOA [11:53:17] ==> INFO: Generating random password for MariaDB
BOA [11:53:17] ==> INFO: Updating MariaDB configuration
BOA [11:53:17] ==> INFO: Restarting MariaDB server...
BOA [11:53:29] ==> INFO: Installing Aegir Master Instance, please wait...
BOA [11:53:30] ==> INFO: MariaDB is listening on 84.201.163.182.
BOA [11:53:30] ==> INFO: Downloading hostmaster modules...
BOA [11:53:42] ==> INFO: Installing drush in /var/aegir
BOA [11:53:46] ==> FATAL ERROR: _PHP_CLI_VERSION must be set to one of supported values
BOA [11:53:46] ==> FATAL ERROR: Aborting AegirSetupM installer NOW!
BOA [11:53:46] ==> FATAL ERROR: AegirSetupM installer failed
BOA [11:53:46] ==> FATAL ERROR: Aborting Barracuda installer NOW!
BOA [11:53:46] ==> HINT: Please check /var/aegir/install.log
BOA [11:53:46] ==> HINT: for more information on errors occured

BOA [11:53:52] ==> Skynet Agent v.BOA-4.0.1 welcomes you aboard!

BOA [11:53:55] ==> ERROR: This octopus installer can be used only when the same version
BOA [11:53:55] ==> of boa or barracuda installer was used before. Your system must be
BOA [11:53:55] ==> upgraded with barracuda installer version BOA-4.0.1 first
BOA [11:53:55] ==> Bye

BOA in-stable completed
Bye

```
